### DISCUSSION TITLE
Buying a beginner board

### DISCUSSION THREAD

**Discussion by bentxt** [toitlang/toit#15] [General]
Hi

As a beginner I'm not sure which board is best to begin with or even compatible at all with toit.

Maybe someone could give some feedback, which of this boards a suitable for toit development.

Thanks in advance
Ben

List:

## DFrobot: Beetle ESP32 
https://www.dfrobot.com/product-1798.html

## DFrobot: FireBeetle ESP32-E 
https://www.dfrobot.com/product-2195.html

## LOLIN D32 ESP32 Board 4 MB FLASH
https://aliexpress.com/item/32808551116.html

## Heltec: Wifi Kit 32
https://heltec.org/project/wifi-kit-32/

## M5stack: ATOM Matrix Esp32
https://shop.m5stack.com/products/atom-matrix-esp32-development-kit

## M5Stack ATOM Lite ESP32 Development Kit
https://shop.m5stack.com/products/atom-lite-esp32-development-kit

## Espressif: ESP32 Pico Kit
https://eu.mouser.com/ProductDetail/Espressif/ESP32-PICO-KIT-V41?qs=sGAEpiMZZMv0NwlthflBixczKuNkn3cUcibEvzahOvM=

**Comment by floitsch** [comment]
From what I can see all models should work with Toit.

The Beetles used to be difficult to flash on macOs because of missing drivers. They are now publicly available here: https://github.com/WCHSoftGroup/ch34xser_macos

I couldn't find the exact spec for the display of the Heltec. It looks like a standard display, so I would expect it to work with Toit. @erikcorry might know more.

**Reply by winksaville** [reply]
What boards have been used during development and testing?

**Reply by floitsch** [reply]
I think I will start taking pictures of all the boards we have at the office, and then start a thread with it.

Currently I'm having six ESP32s on my desk:
- M5Stack Core2: a bit more expensive, but really nice hardware. https://shop.m5stack.com/products/m5stack-core2-esp32-iot-development-kit
- Olimex Esp32: European open-hardware. The printing on the top is extremely hard to read (which is annoying when it's in a breadboard), but the one on the bottom is fine. https://www.olimex.com/Products/IoT/ESP32/ESP32-DevKit-LiPo/open-source-hardware
- ESP32S-board: relatively cheap board. Some users have had difficulties flashing it (needing to press the power button, or needing to add a capacitor): https://randomnerdtutorials.com/solved-failed-to-connect-to-esp32-timed-out-waiting-for-packet-header/. It works for me: https://www.banggood.com/ESP32-Development-Board-WiFi+bluetooth-Ultra-Low-Power-Consumption-Dual-Cores-ESP-32-ESP-32S-Board-p-1109512.html?cur_warehouse=CN
- ESP32_DevKitc_V4: Reference from Espressif. https://www.espressif.com/en/products/devkits/esp32-devkitc
- ESP32 Cam: a bit harder to flash, as it doesn't have an USB-Serial chip. The camera is not yet supported by Toit. https://www.banggood.com/ESP32-CAM-WiFi-+-bluetooth-Camera-Module-Development-Board-ESP32-With-Camera-Module-OV2640-p-1394679.html?cur_warehouse=CN&rmmds=search
- a simple Wroom: https://www.digikey.com/en/products/detail/espressif-systems/ESP32-WROOM-32/8544298
  I'm using a special programmer for it: https://www.banggood.com/ESP32-Test-Board-Burner-Development-Board-WIFI-Module-For-For-ESP-WROOM-32-p-1685003.html?cur_warehouse=CN&rmmds=search

**Reply by skye0402** [reply]
I used a [Wemos D1 R32](https://community.platformio.org/t/wemos-d1-r32-any-users/15304) for development of my project. No problems encountered. Actually Toit seems to be less picky than the Arduino IDE with the many boards you can/ need to choose from.

**Reply by floitsch** [reply]
Thanks. Added it to https://github.com/toitlang/toit/wiki/Devices.
